### PR TITLE
feat: sync chrome-for-testing binary

### PR DESCRIPTION
**Pull request by elrrrrrrr**
> https://github.com/puppeteer/puppeteer/issues/10131 Puppeteer has updated the default browser to Chrome and added the corresponding implementation as follows:

1.  🧶 Added a new category `/-/binary/` for Chrome , exp: `/-/binary/chrome-for-testing/113.0.5672.63/mac-arm64/chrome-mac-arm64.zip`
2. ~~🤖 Replaced the implementation of original chromium version.~~
3. ~~♻️ No longer synchronized the Chromium binary files from past versions of Puppeteer.~~

-----------

> https://github.com/puppeteer/puppeteer/issues/10131 puppeteer 更新了默认浏览器为 chrome，新增对应实现

1.  🧶 `/-/binary/` 新增 chrome binary 分类, 示例链接`/-/binary/chrome-for-testing/113.0.5672.63/mac-arm64/chrome-mac-arm64.zip`
2. ~~🤖 替换原 puppeteer 中 chromium 对应实现~~
3. ~~♻️ 不再同步 puppeteer 历史版本中的 chromium  二进制文件~~

### CODE REVIEW

File: app/common/adapter/binary/ChromeBinary.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { SingletonProto } from '@eggjs/tegg';
+import { BinaryType } from '../../enum/Binary';
+import { AbstractBinary, FetchResult, BinaryItem, BinaryAdapter } from './AbstractBinary';
+
+@SingletonProto()
+@BinaryAdapter(BinaryType.Chrome)
+export class ChromeBinary extends AbstractBinary {
```
Comment (fengmk2) [inline app/common/adapter/binary/ChromeBinary.ts:7]: 品牌名应该叫 ChromeForTesting ，它跟 Chrome 还是有差别的。

`ChromeForTestingBinary`

File: app/common/adapter/binary/ChromeBinary.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        // https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/113.0.5672.63/mac-arm64/chrome-mac-arm64.zip
+        const name = `chrome-${platform}.zip`;
+        this.dirItems[`/${chromeVersion}/${platform}/`] = [
+          {
+            name,
+            date,
+            size: '-',
+            isDir: false,
+            url: `https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/${chromeVersion}/${platform}/${name}`,
+            ignoreDownloadStatuses: [ 404 ],
```
Comment (fengmk2) [inline app/common/adapter/binary/ChromeBinary.ts:62]: 为什么会出现 404？不应该忽略 404

File: app/common/adapter/binary/ChromeForTestingBinary.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      // });
+      const unpkgURL = 'https://unpkg.com/puppeteer-core@latest/lib/cjs/puppeteer/revisions.js';
+      const text = await this.requestXml(unpkgURL);
+      const m = /chrome:\s+\'([\d\.]+)\'\,/.exec(text);
+      if (m) {
+        chromeVersion = m[1];
+      }
+
+      const platforms = [ 'linux64', 'mac-arm64', 'mac-x64', 'win32', 'win64' ];
+      const date = new Date().toISOString();
```
Comment (fengmk2) [inline app/common/adapter/binary/ChromeForTestingBinary.ts:34]: date 每次都是今天，会导致每次做 diff 的时候都会覆盖式下载吗？

Comment (elrrrrrrr) [reply app/common/adapter/binary/ChromeForTestingBinary.ts:34]: diff 目前只根据 name 来判断
这个逻辑和 PuppeteerBinary.ts 获取最新版逻辑一致。[ref](https://github.com/cnpm/cnpmcore/blob/master/app/common/adapter/binary/PuppeteerBinary.ts#L84)

**Review by fengmk2 — APPROVED** [review]

**Comment by fengmk2** [comment]
https://r.cnpmjs.org/binary.html?path=chrome-for-testing/113.0.5672.63/linux64/

**Pull request merged**